week1/HomeWork/index.js

```diff
@@ -0,0 +1,33 @@
+const http = require('http');
+const port = 2000
+let state = 10;
+
+const server = http.createServer((request, response) => {
+
+     console.log(`I have received a request for path: ${ request.url }`);
+     response.writeHead(200, {
+          'Content-Type': 'text/html'
+     });
+     if (request.url === "/") {
+          response.write(`<h1>You requested root path.</h1>`);
+     } else if (request.url === "/state") {
+          response.write(`<h1>You requested the path: ${ request.url }</h1><h2>State: ${state}</h2>`);
+     } else if (request.url === "/add") {
+          state += 1;
+          response.write(`<h1>You requested the path: ${ request.url }</h1><h2>State: ${state}</h2>`);
+     } else if (request.url === "/remove") {
+          state -= 1;
+          response.write(`<h1>You requested the path: ${ request.url }</h1><h2>State: ${state}</h2>`);
+     } else if (request.url === "/reset") {
+          state = 10;
+          response.write(`<h1>You requested the path: ${ request.url }</h1><h2>State: ${state}</h2>`);
+     } else {
+          response.write(`<h1>404 not found</h1>`);
```

Comment:
This will still return a status code of 200 - but rather than just send the text '404', we should return an http status code of 404. (https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)
Can you figure out how to do that? The Node HTTP documentation will help you out.

Reply:
Hi Jason, in the last commit I've added a better solution for the notfound pages, can you check it please?